### PR TITLE
SONAR-7300 Redirect PUT|POST|DELETE api/properties to api/settings

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java [hunk 1]
```diff
@@ -117,4 +110,88 @@ protected String readParam(String key) {
...
+
+    private Optional<String> readParam(String paramKey) {
+      String paramValue = request.getParameter(paramKey);
+      if (Strings.isNullOrEmpty(paramValue)) {
+        return Optional.empty();
+      }
+      return Optional.of(paramValue);
+    }
+
+    private void redirectToReset(Optional<String> key, Optional<String> component) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java]: MINOR Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
MINOR Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)

File: server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java [hunk 2]
```diff
@@ -117,4 +110,88 @@ protected String readParam(String key) {
...
+    }
+
+    void init() {
+      String method = request.getMethod();
+      Optional<String> key = getKey();
+      Optional<String> value = getValue();
+      Optional<String> component = getComponent();
+      switch (method) {
+        case "POST":
+        case "PUT":
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java]: MAJOR Reduce this switch case number of lines from 9 to at most 5, for example by extracting code into methods. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1151)

File: server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java [hunk 3]
```diff
@@ -117,4 +110,88 @@ protected String readParam(String key) {
...
+      return Optional.of(paramValue);
+    }
+
+    private void redirectToReset(Optional<String> key, Optional<String> component) {
+      addParameterIfPresent(PARAM_KEYS, key);
+      addParameterIfPresent(PARAM_COMPONENT, component);
+      redirectedPath = CONTROLLER_SETTINGS + "/" + ACTION_RESET;
+    }
+
+    private void addParameterIfPresent(String parameterKey, Optional<String> value) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java]: MINOR Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 1 major
* MINOR 3 minor

Watch the comments in this conversation to review them.

File: server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java [hunk 4]
```diff
@@ -88,33 +80,124 @@ public void destroy() {
...
+    }
+
+    private void redirectToReset(Optional<String> key, Optional<String> component) {
+      addParameterIfPresent(PARAM_KEYS, key);
+      addParameterIfPresent(PARAM_COMPONENT, component);
+      redirectedPath = CONTROLLER_SETTINGS + "/" + ACTION_RESET;
+      redirectedMethod = "POST";
+    }
+
+    private void addParameterIfPresent(String parameterKey, Optional<String> value) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java]: MINOR Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)

File: server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java [hunk 5]
```diff
@@ -88,33 +84,143 @@ public void destroy() {
...
+          return Optional.empty();
+        }
+        IOUtils.copy(inputStream, writer, UTF_8.name());
+        return Optional.of(writer.toString());
+      } catch (IOException e) {
+        throw new IllegalStateException(e);
+      }
+    }
+
+    private void redirectToReset(Optional<String> key, Optional<String> component) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java]: MINOR Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
MINOR Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)

**Comment by teryk** [comment]
handle api/properties/?…

**Pull request merged**